integrations/outlook/src/index.ts

```diff
@@ -0,0 +1,252 @@
+import { load as loadHtml } from 'cheerio'
+import * as botpress from '.botpress'
+import { GraphApi } from './client'
+import _ from 'lodash'
+
+type Config = botpress.configuration.Configuration
+
+const logger = console
+logger.info('starting integration')
+
+class NotImplementedError extends Error {
+  constructor() {
+    super('Not implemented')
+  }
+}
+
+export default new botpress.Integration({
+  register: async ({ webhookUrl, ctx, client }) => {
+    const graphClient = getClient(ctx.configuration)
+    logger.info('suscribing webhook %s', webhookUrl)
+    const subscriptionId = await graphClient.subscribeWebhook(webhookUrl, ctx)
+    client.setState({
+      type: 'integration',
+      id: ctx.integrationId,
+      name: 'subscriptionInfo',
+      payload: {
+        subscriptionId,
+      },
+    })
+    logger.info('subscriptionId ', subscriptionId)
+
+    /**
+     * This is called when a bot installs the integration.
+     * You should use this handler to instanciate ressources in the external service and ensure that the configuration is valid.
+     */
+  },
+  unregister: async ({ ctx, client }) => {
+    const graphClient = getClient(ctx.configuration)
+
+    const stateRes = await client.getState({
+      id: ctx.integrationId,
+      name: 'subscriptionInfo',
+      type: 'integration',
+    })
+
+    const { state } = stateRes
+    const { subscriptionId } = state.payload
+
+    await graphClient.unsubscribeWebhook(subscriptionId)
+
+    /**
+     * This is called when a bot removes the integration.
+     * You should use this handler to instanciate ressources in the external service and ensure that the configuration is valid.
+     */
+  },
+  actions: {},
+  channels: {
+    channel: {
+      messages: {
+        text: async (props) => {
+          const graphClient = getClient(props.ctx.configuration)
+          await graphClient.sendMail({
+            ...props,
+            body: {
+              contentType: 'text',
+              content: `${props.payload.text}\n\n${props.ctx.configuration.emailSignature}`,
+            },
+          })
+        },
+        image: async () => {
```

Comment:
**issue:** please remove these content types from the definition if you don't intend to implement them